hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/expression/json/JsonQueryFunction.java

```diff
@@ -92,80 +93,89 @@ public HazelcastJsonValue eval(final Row row, final ExpressionEvalContext contex
         // first evaluate the required parameter
         final String path = (String) operands[1].eval(row, context);
         if (path == null) {
-            throw QueryException.error("JSONPath expression can not be null");
+            throw QueryException.error("SQL/JSON path expression cannot be null");
         }
 
         final Object operand0 = operands[0].eval(row, context);
-        final String json = operand0 instanceof HazelcastJsonValue
+        String json = operand0 instanceof HazelcastJsonValue
                 ? operand0.toString()
                 : (String) operand0;
-        if (isNullOrEmpty(json)) {
-            return onEmptyResponse(onEmpty);
+        if (json == null) {
+            json = "";
         }
 
         final JsonPath jsonPath;
         try {
             jsonPath = pathCache.asMap().computeIfAbsent(path, JsonPathUtil::compile);
-        } catch (InvalidPathException exception) {
-            throw QueryException.error("Invalid JSONPath expression: " + exception, exception);
+        } catch (JsonPathCompilerException e) {
+            LOGGER.fine("JSON_QUERY JsonPath compilation failed", e);
```

Comment:
It deserves a comment why we, contrary to the usual practice, we don't chain the cause.

Reply:
Added comments here